Soapuios 356 adding endpoint explorer

ChristinaZelenko opened the pull request:
https://smartbear.atlassian.net/browse/SOAPUIOS-356
adding endpoint explorer

ChristinaZelenko committed:
SOAPUIOS-356
https://smartbear.atlassian.net/browse/SOAPUIOS-356
adding endpoint explorer

ChristinaZelenko committed:
SOAPUIOS-356
https://smartbear.atlassian.net/browse/SOAPUIOS-356
adding new methods for request in endpoint explorer

ChristinaZelenko committed:
Merge branch 'next' of https://github.com/SmartBear/soapui into SOAPUIOS-356-adding-endpoint-explorer

ChristinaZelenko committed:
SOAPUIOS-356
https://smartbear.atlassian.net/browse/SOAPUIOS-356
Refactoring

ChristinaZelenko committed:
Merge branch 'next' of https://github.com/SmartBear/soapui into SOAPUIOS-356-adding-endpoint-explorer

ChristinaZelenko committed:
SOAPUIOS-356
https://smartbear.atlassian.net/browse/SOAPUIOS-356
Adding analytics for the "save request" action in endpoint explorer

IlyaAvdeev commented on soapui/src/main/java/com/eviware/soapui/SoapUI.java:
```diff
@@ -434,6 +451,57 @@ public static void doCommunitySearch(String text) {
+        endpointExplorerButtonPanel = new JPanel(new BorderLayout());
+        endpointExplorerButtonPanel.setPreferredSize(new Dimension(130, 32));
+        endpointExplorerButtonPanel.setMaximumSize(new Dimension(130, 32));
+//        endpointExplorerButtonPanel.setBackground(Color.blue); //TODO: delete
```
do todo

IlyaAvdeev commented on soapui/src/main/java/com/eviware/soapui/impl/rest/actions/explorer/callback/EndpointExplorerCallback.java:
```diff
@@ -0,0 +1,286 @@
+        return headersMap;
+    }
+
+    private void setHeaders(AbstractHttpMessage message, HashMap<String, String> headersMap) {
```
HashMap -> Map it doesn't make sense to show the details of implementation in the method definition

IlyaAvdeev commented on soapui/src/main/java/com/eviware/soapui/impl/rest/actions/explorer/callback/EndpointExplorerCallback.java:
```diff
@@ -0,0 +1,286 @@
+        });
+    }
+
+    private void setHeadersAndPayload(HttpEntityEnclosingRequestBase request, HashMap<String, String> headersMap, String payload) {
```
and here

IlyaAvdeev commented on soapui/src/main/java/com/eviware/soapui/impl/rest/actions/explorer/callback/EndpointExplorerCallback.java:
```diff
@@ -0,0 +1,286 @@
+    private static String getResponseAsString(HttpResponse response) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(response.getStatusLine().toString());
+        builder.append("\r\n");
```
I believe StringUtils has something for this. ("\r\n")

IlyaAvdeev commented on soapui/src/main/java/com/eviware/soapui/impl/rest/actions/explorer/callback/EndpointExplorerCallback.java:
```diff
@@ -0,0 +1,286 @@
+
+        String url = "";
+        String method = "";
+        HashMap<String, String> headersMap = null;
```
HashMap -> Map (the details see below)

IlyaAvdeev commented:
+ please pull the changes from next branch

ChristinaZelenko committed:
SOAPUIOS-356
https://smartbear.atlassian.net/browse/SOAPUIOS-356
Adding analytics for the "save request" action in endpoint explorer

ChristinaZelenko committed:
Merge branches 'SOAPUIOS-356-adding-endpoint-explorer' and 'next' of https://github.com/SmartBear/soapui into SOAPUIOS-356-adding-endpoint-explorer
